internal/pkg/instrumentation/probe/sampling/sampling.go

```diff
@@ -108,3 +188,76 @@ func DefaultParentBasedSampler() ParentBasedConfig {
 		LocalNotSampled:  AlwaysOffID,
 	}
 }
+
+// NewSamplingManager creates a new Manager from the given eBPF collection with the given configuration.
+func NewSamplingManager(c *ebpf.Collection, conf *Config) (*Manager, error) {
+	samplersConfig, ok := c.Maps[samplersConfigMapName]
+	if !ok {
+		return nil, fmt.Errorf("map %s not found", samplersConfigMapName)
+	}
+
+	probeActiveSampler, ok := c.Maps[probeActiveSamplerMapName]
+	if !ok {
+		return nil, fmt.Errorf("map %s not found", probeActiveSamplerMapName)
+	}
+
+	m := &Manager{
+		samplersConfigMap: samplersConfig,
+		ActiveSamplerMap:  probeActiveSampler,
+	}
+
+	err := m.applyConfig(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return m, nil
+}
+
+func (m *Manager) applyConfig(conf *Config) error {
+	samplerIDs := make([]SamplerID, len(conf.Samplers))
+	configs := make([]SamplerConfig, len(conf.Samplers))
+
+	i := 0
+	for id, samplerConfig := range conf.Samplers {
+		samplerIDs[i] = id
+		configs[i] = samplerConfig
+		i++
+	}
```

Comment:
nit: This is a bit confusing about the meaning of `i` and if it shouldn't have been shadowed the following for loop. It may read better if we just append here:

```suggestion
	samplerIDs := make([]SamplerID, 0, len(conf.Samplers))
	configs := make([]SamplerConfig, 0, len(conf.Samplers))

	for id, samplerConfig := range conf.Samplers {
		samplerIDs = append(samplerIDs, id)
		configs = append(configs, samplerConfig)
	}
```

This isn't blocking.